### PR TITLE
Let validator reads key pair from network config and remove --address and --validator_idx

### PR DESCRIPTION
**Pull request by longbowlu**
Since we separate private keys out of `AuthorityPrivateInfo` into a single `KeyPair` in network/genesis config, now we can deterministically know which authority to represent when running a validator. This PR adds this logic to pick `AuthorityPrivateInfo` and removes `--address` and `--validator_idx` as they become unnecessary.

### CODE REVIEW

**Review by oxade — APPROVED** [review]

File: sui/src/bin/validator.rs [hunk 1]
```diff
@@ -62,38 +54,30 @@ async fn main() -> Result<(), anyhow::Error> {
...
     let network_config = match (network_config_path.exists(), cfg.force_genesis) {
         (true, false) => PersistedConfig::<NetworkConfig>::read(&network_config_path)?,
 
         // If network.conf is missing, or if --force-genesis is true, we run genesis.
         _ => {
-            let (network_config, _, _) =
-                genesis(genesis_conf_copy, Some(authority.address)).await?;
+            let genesis_conf: GenesisConfig = PersistedConfig::read(&cfg.genesis_config_path)?;
+            let adddress = SuiAddress::from(genesis_conf.key_pair.public_key_bytes());
+            let (network_config, _, _) = genesis(genesis_conf, Some(adddress)).await?;
```
Comment (oxade) [inline sui/src/bin/validator.rs:64]: Actually, we don't need `address` here
We can infer it from the config
So we should remove the field in genesis method

**Review by oxade — CHANGES_REQUESTED** [review]

File: sui/src/bin/validator.rs (hunk 1 above)
Comment (oxade) [reply sui/src/bin/validator.rs:64]: We can get the address from here https://github.com/MystenLabs/sui/blob/1caf0cce8ec7744395b408e7efc2d5ba55b751eb/sui/src/sui_commands.rs#L388

Comment (longbowlu) [reply sui/src/bin/validator.rs:64]: right we don't. We should clean that up later, trying to get this PR very small and modest

Comment (oxade) [reply sui/src/bin/validator.rs:64]: Dont you think its small enough that we can fix it now?

**Review by huitseeker — APPROVED** [review]
I'd be happy with an issue addressing the cleaning up of the superfluous address calculation, as long as the bench isn't broken here (@oxade  feel free to chime in).

File: sui/src/bin/validator.rs (hunk 1 above)
Comment (longbowlu) [reply sui/src/bin/validator.rs:64]: I don't think it's tiny enough that we should make it here - calls into `genesis`  is spread into multiple files. Also this PR itself does not trigger `address` becoming unused, so they are sorta orthogonal

**Review by oxade — APPROVED** [review]

**Comment by longbowlu** [comment]
@huitseeker  https://github.com/MystenLabs/sui/issues/1758 has context

**Pull request merged**